### PR TITLE
Restoring commit status tracker 

### PR DESCRIPTION
**Pull request by DhanashriPalodkar**
**What type of PR is this?**
Commit Status Tracker was inadvertently disabled during package/command re-org,   This change is to re-enable.  
/kind feature

**What does does this PR do / why we need it**:
This PR add support for commit-status tracker on guthub and gitlab. 

**Which issue(s) this PR fixes**:
Gitops-166
https://issues.redhat.com/browse/GITOPS-166

### CODE REVIEW

File: pkg/odo/cli/pipelines/bootstrap.go [hunk 1]
```diff
@@ -82,9 +82,9 @@ func NewCmdBootstrap(name, fullName string) *cobra.Command {
 
 	initCmd := &cobra.Command{
 		Use:     name,
-		Short:   bootstrapShortDesc,
-		Long:    bootstrapLongDesc,
-		Example: fmt.Sprintf(bootstrapExample, fullName),
+		Short:   initShortDesc,
```
Comment (chetan-rns) [inline pkg/odo/cli/pipelines/bootstrap.go]: We don't have to change these to `init`. I guess it happened while copying files

**Review by wtam2018 — COMMENTED** [review]
There is a failed test.

FAIL    github.com/openshift/odo/pkg/pipelines  0.713s

File: pkg/pipelines/triggers/pipelinerun_test.go [hunk 2]
```diff
@@ -82,32 +83,4 @@ func TestCreateStageCIPipelineRun(t *testing.T) {
 	if diff := cmp.Diff(validStageCIPipeline, template); diff != "" {
 		t.Fatalf("createCIPipelineRun failed:\n%s", diff)
 	}
-}
-
-func TestCreateDevResource(t *testing.T) {
```
Comment (wtam2018) [inline pkg/pipelines/triggers/pipelinerun_test.go]: Why this test is deleted?

File: pkg/pipelines/triggers/templates.go [hunk 3]
```diff
@@ -48,10 +48,8 @@ func CreateDevCDDeployTemplate(ns, saName string) triggersv1.TriggerTemplate {
 // CreateDevCIBuildPRTemplate creates DevCIBuildPRTemplate
 func CreateDevCIBuildPRTemplate(ns, saName string) triggersv1.TriggerTemplate {
 	return triggersv1.TriggerTemplate{
-		TypeMeta: triggerTemplateTypeMeta,
-		ObjectMeta: meta.ObjectMeta(
-			meta.NamespacedName(ns, "app-ci-template"),
-			statusTrackerAnnotations("dev-ci-build-from-pr", "Dev CI Build")),
```
Comment (wtam2018) [inline pkg/pipelines/triggers/templates.go]: You deleted this fix for adding annotation.

File: pkg/pipelines/init.go [hunk 4]
```diff
@@ -180,7 +179,7 @@ func createCICDResources(fs afero.Fs, repo scm.Repository, pipelineConfig *confi
 	githubSecret, err := secrets.CreateSealedSecret(meta.NamespacedName(cicdNamespace, eventlisteners.GitOpsWebhookSecret),
 		gitOpsWebhookSecret, eventlisteners.WebhookSecretKey)
 	if err != nil {
-		return nil, fmt.Errorf("failed to generate GitHub Webhook Secret: %v", err)
+		return nil, fmt.Errorf("failed to generate GitHub Webhook Secret: %w", err)
```
Comment (wtam2018) [inline pkg/pipelines/init.go]: Please revert this.

File: pkg/pipelines/service.go [hunk 5]
```diff
@@ -132,29 +132,26 @@ func serviceResources(m *config.Manifest, fs afero.Fs, p *AddServiceParameters)
...
 	resources = res.Merge(svcImageBinding, resources)
-	filenames = append(filenames, bindingFilename)
-
+	
 	if isInternalRegistry {
-		files, regRes, err := imagerepo.CreateInternalRegistryResources(cfg, roles.CreateServiceAccount(meta.NamespacedName(cfg.Name, saName)), imageRepo)
+		regRes, err := imagerepo.CreateInternalRegistryResources(cfg, roles.CreateServiceAccount(meta.NamespacedName(cfg.Name, saName)), imageRepo)
 		if err != nil {
-			return nil, nil, "", fmt.Errorf("failed to get resources for internal image repository: %v", err)
+			return nil, "", fmt.Errorf("failed to get resources for internal image repository: %w", err)
```
Comment (wtam2018) [inline pkg/pipelines/service.go]: Please revert this %w -> %v.

File: pkg/pipelines/init.go [hunk 6]
```diff
@@ -132,11 +132,11 @@ func CreateDockerSecret(fs afero.Fs, dockerConfigJSONFilename, ns string) (*ssv1
 
 	authJSONPath, err := homedir.Expand(dockerConfigJSONFilename)
 	if err != nil {
-		return nil, fmt.Errorf("failed to generate path to file: %v", err)
+		return nil, fmt.Errorf("failed to generate path to file: %w", err)
```
Comment (wtam2018) [inline pkg/pipelines/init.go]: Please revert.

File: pkg/pipelines/init.go [hunk 7]
```diff
@@ -132,11 +132,11 @@ func CreateDockerSecret(fs afero.Fs, dockerConfigJSONFilename, ns string) (*ssv1
...
 	}
 	f, err := fs.Open(authJSONPath)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read docker file '%s' : %v", authJSONPath, err)
+		return nil, fmt.Errorf("failed to read docker file '%s' : %w", authJSONPath, err)
```
Comment (wtam2018) [inline pkg/pipelines/init.go]: Please revert.

File: pkg/pipelines/triggers/templates.go [hunk 8]
```diff
@@ -98,13 +95,12 @@ func CreateCDPushTemplate(ns, saName string) triggersv1.TriggerTemplate {
 // CreateCIDryRunTemplate returns TriggerTemplate for CI Dry Try
 func CreateCIDryRunTemplate(ns, saName string) triggersv1.TriggerTemplate {
 	return triggersv1.TriggerTemplate{
-		TypeMeta: triggerTemplateTypeMeta,
-		ObjectMeta: meta.ObjectMeta(meta.NamespacedName(ns, "ci-dryrun-from-pr-template"),
-			statusTrackerAnnotations("ci-dryrun-from-pr-pipeline", "Stage CI Dry Run")),
```
Comment (wtam2018) [inline pkg/pipelines/triggers/templates.go:103]: You lost this fix.   I think you copied you files on top of these files.  So, the fixes that are done before yours are lost.  Please go back to check the diffs to make sure the previous fixes are preserved.

**Review by wtam2018 — CHANGES_REQUESTED** [review]
There are some issues need to be addressed.   I'd like to do an e2e test after issues are resolved.

**Comment by wtam2018** [comment]
--status-tracker-access-token should be available to "pipelines init".   I should be able to enable commit status tracker when I run "odo pipelines init".

File: pkg/odo/cli/pipelines/bootstrap.go (hunk 1 above)
Comment (DhanashriPalodkar) [reply pkg/odo/cli/pipelines/bootstrap.go]: yes I have fixed it.. Pushing it in next commit

File: pkg/pipelines/init.go (hunk 4 above)
Comment (DhanashriPalodkar) [reply pkg/pipelines/init.go]: yes

File: pkg/pipelines/triggers/templates.go (hunk 8 above)
Comment (DhanashriPalodkar) [reply pkg/pipelines/triggers/templates.go:103]: sure, on it

**Pull request closed without merging**